subsys/shell/shell.c

```diff
@@ -1663,6 +1663,21 @@ void shell_help_print(const struct shell *shell,
 	help_subcmd_print(shell);
 }
 
+int shell_prompt_change(const struct shell *shell, char *prompt)
+{
+
+	size_t len = shell_strlen(prompt);
```

Comment:
isn't strlen returning length without \0. If so then if check should be
`len < CONFIG...` and memcpy `len+1`

Reply:
It is fine, prompt table is defined as: `[CONFIG_SHELL_PROMPT_LENGTH + 1]`

Reply:
ok, but then you are missing setting termination ('\0'). You should have `len+1` in memcpy

Reply:
@nordic-krch : done